### PR TITLE
Fix:  issue 774 [fix metadata default port]

### PR DESCRIPTION
**Pull request by sanxun0325**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #
https://github.com/apache/dubbo-go/issues/774

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by zouyx — COMMENTED** [review]
Merge develop and make Checks to normal.

**Review by zouyx — COMMENTED** [review]
Merge develop and make Checks to normal

File: metadata/service/exporter/configurable/exporter_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@
 package configurable
 
 import (
+	"github.com/apache/dubbo-go/common"
```
Comment (AlexStocks) [inline metadata/service/exporter/configurable/exporter_test.go:21]: move it to the third import block.

File: metadata/service/exporter/configurable/exporter.go [hunk 2]
```diff
@@ -46,12 +46,21 @@ func NewMetadataServiceExporter(metadataService service.MetadataService) exporte
...
-		serviceConfig.Protocols = map[string]*config.ProtocolConfig{
-			constant.DEFAULT_PROTOCOL: generateMetadataProtocol(),
+		if url != nil && url.SubURL != nil {
+			serviceConfig.Protocols = map[string]*config.ProtocolConfig{
+				constant.DEFAULT_PROTOCOL: {
+					Name: url.SubURL.Protocol,
+					Port: url.SubURL.Port,
+				},
+			}
+		} else {
```
Comment (hxmhlt) [inline metadata/service/exporter/configurable/exporter.go:60]: this else is not necessary

**Review by AlexStocks — APPROVED** [review]

File: metadata/service/exporter/configurable/exporter.go (hunk 2 above)
Comment (sanxun0325) [reply metadata/service/exporter/configurable/exporter.go:60]: When the URL is nil, it returns an error?

Comment (hxmhlt) [reply metadata/service/exporter/configurable/exporter.go:60]: Yes

Comment (hxmhlt) [reply metadata/service/exporter/configurable/exporter.go:60]: yes

File: metadata/service/exporter/configurable/exporter.go [hunk 3]
```diff
@@ -46,12 +47,18 @@ func NewMetadataServiceExporter(metadataService service.MetadataService) exporte
...
 
 // Export will export the metadataService
-func (exporter *MetadataServiceExporter) Export() error {
+func (exporter *MetadataServiceExporter) Export(url *common.URL) error {
 	if !exporter.IsExported() {
 		serviceConfig := config.NewServiceConfig(constant.SIMPLE_METADATA_SERVICE_NAME, context.Background())
 		serviceConfig.Protocol = constant.DEFAULT_PROTOCOL
+		if url == nil || url.SubURL == nil {
+			return errors.New("the url or url.SubURL cannot be nil")
+		}
```
Comment (hxmhlt) [inline metadata/service/exporter/configurable/exporter.go:56]: Pls modify the error description. Such as 'Metadata Server url is nil, pls check your configuration'. It should be friendly infomation.

Comment (zouyx) [reply metadata/service/exporter/configurable/exporter.go:56]: @sanxun0325  fix or not?

Comment (sanxun0325) [reply metadata/service/exporter/configurable/exporter.go:56]: @zouyx updated

**Review by zouyx — APPROVED** [review]

**Review by hxmhlt — APPROVED** [review]

**Pull request merged**